Mathlib/GroupTheory/Congruence.lean

```diff
@@ -1176,32 +1174,40 @@ instance {M : Type*} [Monoid M] (c : Con M) : Pow c.Quotient ℕ where
 @[to_additive "The quotient of an `AddSemigroup` by an additive congruence relation is
 an `AddSemigroup`."]
 instance semigroup {M : Type*} [Semigroup M] (c : Con M) : Semigroup c.Quotient :=
-  Function.Surjective.semigroup _ Quotient.surjective_Quotient_mk'' fun _ _ => rfl
```

Comment:
I think a note explaining situation is good here. I would hope the slowdown in unification we see here (and in related files) can be fixed further up the file.  Similarly for the other explicit projections below.